simulation/adx_gyro_accelerometer/src/main/native/include/ADXRS450_SpiGyroWrapperData.h

```diff
@@ -32,12 +34,17 @@ class ADXRS450_SpiGyroWrapper {
   void SetAngle(double angle);
 
  private:
-  const int m_port;
+  int m_port;
+  int m_readCallbackId;
+  int m_autoReceiveReadCallbackId;
 
   wpi::mutex m_registerMutex;
   std::atomic<double> m_angle{0.0};
+  std::atomic<double> m_angleDiff{0.0};
```

Comment:
It's worth noting that std::atomic<double> is not lock-free.  Would it be better to have an explicit lock for angle and angleDiff?

Reply:
I think so. Here's some assembly from godbolt.org: https://godbolt.org/g/984MDM. There's lots of redundant register shuffling from the implicit loads and stores for the atomic. According to the data in https://youtu.be/9hJkWwHDDxs, atomics really shine for high numbers of threads and high contention workloads. (That specific data is for an atomic increment instruction behind std::atomic, which this code is not). This code will probably be called by 3 threads at most. The mutex version being much easier to reason about is a bonus.

Reply:
I thought about doing that from the jump when I refactored this, but thought I should keep it more like the other mock stuff.  Updated to explicit mutex